### PR TITLE
Additional tests including leaq, setb, cmp, redefined_sym, add

### PR DESCRIPTION
**Pull request by thbwd**
Some additional tests I wrote yesterday. ✌️

### CODE REVIEW

File: 02/studenttests.ml [hunk 1]
```diff
@@ -256,13 +261,25 @@ let cc_sub = fun (dest:Int64.t) (src:Int64.t) -> Gradedtests.test_machine
...
+  ;InsB0 (Negq, [~%Rax]);InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag
+  ]
+
 let condition_flag_set_tests = 
   [ (* Logic instruction. *)
     ("cc_and", Gradedtests.cs_test 2 (cc_and ()) (false, false, false))
   ; ("cc_or",  Gradedtests.cs_test 2 (cc_or  ()) (false, false, false))
   ; ("cc_xor", Gradedtests.cs_test 2 (cc_xor ()) (false, false, false))
   ; ("cc_not", Gradedtests.csi_test 2 (cc_not ()))
-  ; ("cc_not_negative", Gradedtests.cs_test 3 (cc_not_negative ()) (false, false, false))
```
Comment (nicdard) [inline 02/studenttests.ml:265]: Why are you removing this test? Is this covered by a new one? If this is the case please remove also the `cc_not_negative` definition

Comment (nicdard) [reply 02/studenttests.ml:265]: Please use cc as prefix name, as all the other cases

File: 02/studenttests.ml [hunk 2]
```diff
@@ -256,13 +261,25 @@ let cc_sub = fun (dest:Int64.t) (src:Int64.t) -> Gradedtests.test_machine
...
+  ; ("cs_not", Gradedtests.csi_test 2 (cc_not ()))
+  ; ("cc_and_arg =0", Gradedtests.cs_test 2 (cc_and_arg 0 12345) (false, false, true))
+  ; ("cc_and_arg =-1", Gradedtests.cs_test 2 (cc_and_arg (-1) (-1)) (false, true, false))
```
Comment (nicdard) [inline 02/studenttests.ml:282]: Can you use snake case and avoid spaces? Just for consistency
You may use eq for = and neg for -

File: 02/studenttests.ml [hunk 3]
```diff
@@ -481,19 +518,64 @@ let instruction_tests = [
       && int64_of_sbytes (Gradedtests.sbyte_list m.mem (mem_size-8)) = Int64.sub Int64.max_int 254L
     )
   );
+  ("setb_cmp", Gradedtests.machine_test "1" 6 (set_cmp 10L 10L Eq)
```
Comment (nicdard) [inline 02/studenttests.ml:521]: Can you use a description of the expected output as in the other test cases? Also I think some tests that follow contain typos in the description

File: 02/studenttests.ml [hunk 4]
```diff
@@ -481,19 +518,64 @@ let instruction_tests = [
...
     (fun m -> m.regs.(rind Rax) = -20L));
+  ("leaq", Gradedtests.machine_test "90()" 2 (leaq_simpl 0L (Ind1 (Lit 90L)))
+    (fun m -> m.regs.(rind Rax) = 90L));
+  ("leaq", Gradedtests.machine_test "$90" 2 (leaq_simpl 10L (Ind2 Rax))
+    (fun m -> m.regs.(rind Rax) = 10L));
+  ("leaq", Gradedtests.machine_test "$90" 2 (leaq_simpl 10L (Ind3 (Lit 10L, Rax)))
+    (fun m -> m.regs.(rind Rax) = 20L));
 ]
 
+let redefinedsym_test (p:prog) () =
```
Comment (nicdard) [inline 02/studenttests.ml:561]: Please remove it, it's already defined at the very beginning

File: 02/studenttests.ml [hunk 5]
```diff
@@ -481,19 +518,64 @@ let instruction_tests = [
...
+    | _ -> failwith "Should have raised Redefined_sym"
+
 let provided_tests : suite = [
   Test ("Debug: End-to-end Tests", [
     ("empty main", Gradedtests.program_test [empty_main] 0L)
     ; ("negq", Gradedtests.program_test (main_driver::negq) (Int64.neg 99L) )
     ; ("addq", Gradedtests.program_test (main_driver::addq) 3L )
     ; ("fibonacci", Gradedtests.program_test (main_driver::fibonacci) 832040L)
   ]);
+  Test ("Debug: Condition Flags Set Tests", [
```
Comment (nicdard) [inline 02/studenttests.ml:575]: We already have the sale group, can you merge them or change this name if the group Is different,

File: 02/studenttests.ml [hunk 6]
```diff
@@ -481,19 +518,64 @@ let instruction_tests = [
...
+    ("redefined sym", redefinedsym_test [text "main" [Retq,[]]; text "main" [Retq,[]]])
```
Comment (nicdard) [inline 02/studenttests.ml:576]: Probably those should go into e2e or create a new assemble category since the errore is thrown there

**Review by nicdard — CHANGES_REQUESTED** [review]
LGTM, some minor comments

File: 02/studenttests.ml (hunk 2 above)
Comment (thbwd) [reply 02/studenttests.ml]: I've changed it, but I personally do not find these names to be very helpful. For example, seeing that "cc_sar_0_a: OF:false SF:false ZF:false"  failed (randomly picked) does not tell me what's really going on without reading the test code anyway. In my experience, BDD leads to more understandable test descriptions, but that might be overkill for these projects.

**Pull request merged**